src/global/project/environment.rs

```diff
@@ -0,0 +1,72 @@
+use std::{borrow::Borrow, fmt, str::FromStr};
+
+use miette::Diagnostic;
+use regex::Regex;
+use serde::{self, Deserialize, Deserializer};
+use thiserror::Error;
+
+/// Represents the name of an environment.
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub(crate) struct EnvironmentName(String);
+
+impl EnvironmentName {
+    /// Returns the name of the environment.
+    pub fn as_str(&self) -> &str {
+        &self.0
+    }
+}
+
+impl Borrow<str> for EnvironmentName {
+    fn borrow(&self) -> &str {
+        self.as_str()
+    }
+}
+
+impl fmt::Display for EnvironmentName {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl PartialEq<str> for EnvironmentName {
+    fn eq(&self, other: &str) -> bool {
+        self.as_str() == other
+    }
+}
+
+impl<'de> Deserialize<'de> for EnvironmentName {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let name = String::deserialize(deserializer)?;
+        EnvironmentName::from_str(&name).map_err(serde::de::Error::custom)
```

Comment:
I think we can call name.parse() directly as we already implemented FromStr

Reply:
Yes, that works!